yapftests/yapf_test.py

```diff
@@ -1436,5 +1436,247 @@ def testSimple(self):
     self._Check(unformatted_code, expected_formatted_code)
 
 
+class HorizontallyAlignedTrailingCommentsTest(unittest.TestCase):
```

Comment:
Please add some examples with disabled lines.

Reply:
Unfortunately, I'm not sure what you mean by disabled lines. Do you mean tests where spaces_before_comment is set to an integer rather than a list?

Reply:
Lines which are within a disabled section. Either they have a `# yapf: disable` comment on the line or around the line, or they're in a section not covered by the `--lines` option.

So something like:

```
# yapf: disable
func()  # foo
b() # b
# yapf: enable
```

None of those comments should be modified. Similarly:

```
func()  # yapf: disable
b() # a comment
g() # align with b()'s comment
```

In this `func()`'s comment shouldn't be aligned, but `b()`'s and `g()`'s should. (At least if I understand what you're going for with this feature. :-)

Reply:
Ah, got it. You are right in that this change as it is now will not respect this behavior. I'll add some tests and implement the functionality.

Thanks!